libs/MotionKit/source/MotionKit.cpp

```diff
@@ -58,56 +56,54 @@ void MotionKit::startStabilisation()
 {
 	stop();
 
-	_imukit.start();
-	_imukit.setOrigin();
+	_previous_euler_angles = _imukit.getEulerAngles();
+	_pid.setTargetYaw(_previous_euler_angles.yaw);
 
 	_target_not_reached		  = false;
 	_stabilisation_requested  = true;
 	_rotate_x_turns_requested = false;
 
-	_event_loop.start();
+	auto on_euler_angles_rdy_callback = [this] { run(); };
+
+	_imukit.onEulerAnglesRdy(on_euler_angles_rdy_callback);
 }
 
 // LCOV_EXCL_START - Dynamic behavior, involving motors and time.
 void MotionKit::run()
 {
-	auto last_yaw			= kReferenceAngle;
-	auto rotations_executed = 0;
+	auto must_rotate = [&] { return _rotate_x_turns_requested && _rotations_executed < _rotations_to_execute; };
+	auto motion_end	 = [&] { return !_stabilisation_requested && !_rotate_x_turns_requested && !_target_not_reached; };
```

Comment:
I'd keep the same structure `must_rotate` vs `must_stop`

```suggestion
	auto must_rotate = [&] { return _rotate_x_turns_requested && _rotations_executed < _rotations_to_execute; };
	auto must_stop	 = [&] { return !_stabilisation_requested && !_rotate_x_turns_requested && !_target_not_reached; };
```